contributor-docs/adrs/adr-xxx-behavior-isolation.md

```diff
@@ -0,0 +1,137 @@
+
+Common in React applications is also the use of centralised state management. Centralised state management tools handle in memory data, and can dispatch it to arbitrary points in a React tree, which can then be mapped to props.
+
+ADR 002 claimed that React Hooks offer better extensibility as they can interact with state which is driven by the consumer, while Custom Elements require listening to events on the document.
+
+Given state management libraries are agnostic to the underyling framework, it is important to acknowledge that it is entirely possible to use state management libraries such as Redux or Mobx with many component frameworks such as Vue, Svelte or React, as well as with Custom Elements.
+
+React components may also directly communicate with Custom Elements using the same mechanisms that React components communicate with built in elements - by passing props within JSX. A complication arises with React as it hard codes a list of attributes that map to class properties, and a list of `on*` prefixed callbacks that map to events. For example using the `hidden` prop on element rendered by React, react will look up the `hidden` property in its hard coded loookup table and map the prop to a `.hidden = ` call. Props given to JSX elements which are not in this hard coded table map to `setAttribute(name)` calls, and the result is stringified.
+
+While React's support for custom events and custom attributes is not as robust as, say, Preact's, it is entirely possible to map attributes and events to React props.
+
+Today, Primer React includes the `RelativeTime` component. Using the `@lit-labs/react` library `RelativeTime` wraps the `relative-time` custom element and automatically maps props directly to the class instance, and converts `on*` callback props to event listeners that can then observe event announcements from the element. In this way, `RelativeTime` behaves just like if it were written in React. The utilisation of the `@lit-labs/react` library requires 1 line of code:
+
+```typescript
+const RelativeTime = styled(createComponent(React, 'relative-time', RelativeTimeElement))(sx)
+```
+
+While React 18 and below require a small library like `@lit-labs/react`, due to the [lack of support for Custom Element](https://custom-elements-everywhere.com/libraries/react/results/results.html), React 19 [will likely need no such library](https://custom-elements-everywhere.com/libraries/react-experimental/results/results.html).
+
+
+#### Organizational overhead
```

Comment:
Another concern for this section is the fact that only one version of a custom element (by tag name) can be registered on a page at a time. In dotcom, this will be a particular issue if you are soft navigating between pages using PVC and PRC, which could be using _different versions_ of the underlying custom elements. If there were a breaking change to the custom element, both PRC and PVC will need to be updated at the same time.

In contrast, if we used vanilla JS behaviors and PRC/PVC got out of sync for a short period of time, the client is able to run _both_ versions of the behavior in the same page. This obviously isn't ideal from a bundle-size perspective, but it does ensure we do not accidentally release a breaking change to our users.